assets/javascripts/modules/autoCompleteFactory.js

```diff
@@ -2,15 +2,37 @@ require('jquery');
 var autoComplete = require('./autoComplete.js');
 
 var createAutoCompleteCountries = function () {
-  var $chooseCountryAutoCompleteElem = $('.js-choose-country-auto-complete');
-  var $countryCodeInput = $('#countryCode');
-  var suggestionDisplayTemplate = function (title, value) {
-    return title + ' (+' + value + ')';
-  };
+  $('.js-choose-country-auto-complete').each(function(){
+    
+    var $countryCodeInput;
+    if($('#countryCode').length > 0){
+      $countryCodeInput = $('#countryCode');
+    } else {
+      $countryCodeInput = $('[id="' + $(this).attr("data-suggestions") + '"]');    
+    }
```

Comment:
> I wanted to allow $countryCodeInput to be definable from the view as it is currently hard-coded in the factory. If the ID is already defined in the view (or the list is not one of countries) we need to allow this (using a defined ID also restricts instance use).

Agreed. So I think we are saying the same thing here. Send in the `$countryCodeInput` via a data attribute. The `#countryCode` just needs to be kept for backwards compatibility. 
I see a future bit of work to remove the `#countryCode` selector from the `Js` and move it into the `data-` attribute in the view for services who have not refactored to the `data-` way.

As said above for this check it doesn't feel right to have it in the factory, this can also go into the `autoComplete.js`